### DISCUSSION TITLE
Improve project overview

### DISCUSSION THREAD

**Discussion by vjpr** [oven-sh/bun#1503] [Ideas]
**tldr**: We should organize the roadmap better using Github projects and issue labels.

# Problem

It's difficult to see an overview of the project for future users and new contributors.

Currently there are two big umbrella issues: 

- https://github.com/oven-sh/bun/issues/159
- https://github.com/oven-sh/bun/issues/798

The problem with umbrella issues is they are not collaborative. As seen by people posting big lists of issues in the "Bun's priorities" issue.

I think most people want to know what is the list of missing features before I can run my favorite tool (next.js, cra, webpack, etc.) or library (chalk). New contributors will get involved because they want to push support for a certain tool over the line. Better project organization will help identify what needs to be done.

There are already a lot of issues being created for "Support X tool".

And this is already a stated goal:

> Simplify contribution. The vast majority of Bun's code was written by me. More people need to be involved.
> From: https://github.com/oven-sh/bun/issues/798

# Proposal

## New label: `eco:<tool>`

For important tools, we create a single umbrella issue e.g. "Support Next.js" labelled "ecosystem". This is for general discussion. It has the following content:

```
# Support Next.js

Task List: https://github.com/orgs/oven-sh/projects/2/views/3
```

Any tasks related to this tool should be labelled with "eco:next.js" or "eco:next.js:v13".

## Github project: `Bun Ecosystem`

We then create a Github Project called "Bun Ecosystem", with a view for each tool filtering on these labels. We could use two views, one for those labelled with `bug` to collect all the bug submissions for triage, and another for tracking features.

![Screenshot 2022-11-14 at 2 52 23 pm](https://user-images.githubusercontent.com/281413/201677351-c95dc777-53ff-410f-9f36-f5ebd379bcc5.png)

**Comment by vjpr** [comment]
Further, we could create views for each of the project areas (from https://github.com/oven-sh/bun#not-implemented-yet):

The existing labels could use another level of categorization. E.g:

- component:transpiler
- component:transpiler:minifier
- component:transpiler:jsx
- component:transpiler:ts
- component:package-manager
- component:bundler
- component:css
- component:css:parser
- component:runtime

---

Could also be used to track parity with Node.js APIs.

**Comment by melroy89** [comment]
Actually try to add GitHub Milestones.. That would help alot.
